chromium_src/components/update_client/request_sender.cc

```diff
@@ -33,8 +33,12 @@ class BraveEcdsa : public Ecdsa {
  public:
   static std::unique_ptr<Ecdsa> Create(int key_version,
                                        const std::string_view& public_key) {
-    std::string base64_decoded = GetKey(kBraveKeyPubBytesBase64);
-    return Ecdsa::Create(kBraveKeyVersion, base64_decoded);
+    if (brave::ShouldUseOmaha4()) {
+      const std::string base64_decoded = GetKey(kBraveKeyPubBytesBase64);
```

Comment:
why is the public key different for omaha 4? How could this ever work using the public key in `kKeyPubBytesBase64` (non-omaha)? This is also changing behavior for non-omaha 4 isn't it?

Reply:
but doesn't this change the behavior when O4 is not enabled? It was using `kBraveKeyVersion` before so why would that change here? Isn't this going to break sparkle because it's trying to use the chromium key?

Reply:
It only used `kBraveKeyVersion` when `BRAVE_ENABLE_UPDATER`. This build flag was the preliminary O4 flag and was never enabled. So I don't think it was using `kBraveKeyVersion` before.

Reply:
The only build flag I see here is `IS_MAC`. Is this the one you mean?

Reply:
per dm discussion we should remove the IS_MAC buildflag as os buildflags shouldn't be used as a substitute for feature flags, but since cup signing is disabled we can either remove this override completely or add a NOTREACHED_NORETURN